cpp/src/io/utilities/type_conversion.cu

```diff
@@ -34,12 +34,15 @@ data_type convert_string_to_dtype(const std::string& dtype_in)
     return static_cast<char>(std::tolower(ch));
   });
   if (dtype == "str") return data_type(cudf::type_id::STRING);
-  if (dtype == "timestamp[s]") return data_type(cudf::type_id::TIMESTAMP_SECONDS);
+  if (dtype == "timestamp[s]" || dtype == "datetime64[s]")
```

Comment:
We should really just get rid of this entire thing and make the C++ have users provide libcudf type instances instead of strings. Can we add a TODO comment at the callsite of this function?

Reply:
Added a todo in this function as we seem to be calling this in two different code-flows.